R/gen_data.R

```diff
@@ -1,53 +1,159 @@
-#' Generate dummy datasets
-#' This function produces randomised datasets based on MSF data dictionaries.
+#' MSF data dictionaries and dummy datasets
+#' This function reads in MSF data dictioanries and produces randomised datasets based these
 #' @param disease Specify which disease you would like to use.
 #' Currently supports "Cholera", "Measles" and "Meningitis".
 #' @param varnames Specify name of column that contains varnames.
 #' Currently default set to "Item".
 #' (this can probably be deleted once dictionaries standardise)
-#' @param numcases Specify the number of cases you want (default is 300)
+#' @param numcases For fake data, specify the number of cases you want (default is 300)
+#' @param tibble Return data dictionary as a tidyverse tibble (default is TRUE)
 #' @import rio
 #' @import epitrix
-#' @noRd
+#' @import dplyr
+#' @export
+
+
+# function to
+msf_dict <- function(name = "MSF-outbreak-dict.xls", disease, tibble = TRUE) {
+
+  # get excel file path (need to specify the file name)
+  path <- system.file("extdata", name, package = "sitrep")
 
-gen_data <- function(disease, varnames = "Item", numcases = 300) {
   # read in data set - pasting the disease name for sheet
-  dat_dict <- rio::import(epidict(), which = paste0(disease, "_content"))
+  dat_dict <- rio::import(path, which = paste0(disease, "_content"))
+
+  # clean col names
+  colnames(dat_dict) <- epitrix::clean_labels(colnames(dat_dict))
+
+  # clean future var names (shortname_export)
+  dat_dict$shortname_export <- epitrix::clean_labels(dat_dict$shortname_export)
+
+  # DATA CLEANING
+  # (CAN EVENTUALLY BE DELETED WHEN DICTIONARIES STABILISE)
+  # 3 types added to - BOOLEAN, TRUE_ONLY and ORGANISATION-UNIT
+  dat_dict$options[which(is.na(dat_dict$options) &
```

Comment:
You don't actually need the "which" function here; the logical subsetting will take care of things